org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -293,4 +311,39 @@ public static String getVersion() {
 		return context == null? "Unknown":context.getBundle().getVersion().toString();
 	}
 
+	private static void redirectStandardStreams() throws FileNotFoundException {
+		in = System.in;
+		out = System.out;
+		err = System.err;
+		System.setIn(new ByteArrayInputStream(new byte[0]));
+		boolean isDebug = Boolean.getBoolean("jdt.ls.debug");
+		if (isDebug) {
+			String id = ("ls-" + new Timestamp(System.currentTimeMillis())).replace(" ", "_");
```

Comment:
I'd rather [avoid filenames containing slashes](https://stackoverflow.com/questions/9847288/is-it-possible-to-use-in-a-filename) in them, like `.error-ls-2017-05-26_12/36/09.998.log`